src/HW_models/NRF_DPPI.c

```diff
@@ -0,0 +1,224 @@
+{
+  dest_f_t task_function;
+  struct subscr_list_elem * p_next;
+} subscr_list_elem_t;
+
+//Table with TASKs subscribing to all channels
+static subscr_list_elem_t* dppi_subscriber_list[NUMBER_DPPI_CHANNELS];
+
+//Function for checking if an element is in the subscriber linked list
+static bool subscriber_list_check(dest_f_t task_function, subscr_list_elem_t ** pp_head)
+{
+  if (*pp_head == NULL)
+  {
+    return false;
+  }
+
+  subscr_list_elem_t *p_last_elem = *pp_head;
+  bool in_list = (p_last_elem->task_function == task_function);
+
+  while (p_last_elem->p_next != NULL)
+  {
+    if (p_last_elem->task_function == task_function)
+    {
+      in_list = true;
+    }
+    p_last_elem = p_last_elem->p_next;
+  }
+
+  return in_list;
+}
+
+//Function for appending an element to the subscriber linked list
+static void subscriber_list_append(subscr_list_elem_t ** pp_head, dest_f_t task_function)
+{
+  if (!subscriber_list_check(task_function, pp_head))
+  {
+    struct subscr_list_elem* p_elem = (struct subscr_list_elem*) malloc(sizeof(struct subscr_list_elem));
+    p_elem->task_function = task_function;
+    p_elem->p_next = (*pp_head);
+    *pp_head = p_elem;
+  }
+}
+
+//Function for removing an element from the subscriber linked list
+static void subscriber_list_remove(subscr_list_elem_t ** pp_head, dest_f_t task_function)
+{
+  if(subscriber_list_check(task_function, pp_head))
+  {
+    struct subscr_list_elem* temp = (struct subscr_list_elem*) malloc(sizeof(struct subscr_list_elem));
+    temp = *pp_head;
+    subscr_list_elem_t *prev;
+
+    if (temp != NULL && temp->task_function == task_function)
+    {
+      *pp_head = temp->p_next;
+      free(temp);
+      return;
+    }
+
+    while (temp != NULL && temp->task_function != task_function)
+    {
+      prev = temp;
+      temp = temp->p_next;
+    }
+
+    if (temp == NULL)
+    {
+      free(temp);
+      return;
+    }
+
+    prev->p_next = temp->p_next;
+    free(temp);
+  }
+}
+
+/**
+ * Update the subscriber list by adding the task to the specified channel
+ */
+void nrf_dppi_subscriber_add(uint8_t channel, dest_f_t function)
+{
+  subscriber_list_append(&dppi_subscriber_list[channel], function);
+}
+
+/**
+ * Update the subscriber list by removing the task from all channels
+ */
+void nrf_dppi_subscriber_remove(dest_f_t function)
+{
+  for(int i=0; i < NUMBER_DPPI_CHANNELS; i++)
+  {
+    subscriber_list_remove(&dppi_subscriber_list[i], function);
+  }
+}
+
+/**
+ * Publish to a channel by triggering all tasks subscribing to the channel
+ */
+void nrf_dppi_publish(uint8_t channel)
+{
+  struct subscr_list_elem* temp = (struct subscr_list_elem*) malloc(sizeof(struct subscr_list_elem));
+  temp = dppi_subscriber_list[channel];
+  subscr_list_elem_t *prev;
+
+  while (temp != NULL)
+  {
+    temp->task_function();
+    prev = temp;
+    temp = temp->p_next;
+  }
+
+  free(temp);
+}
+
+
+/**
+ * Initialize the DPPI model
+ */
+void nrf_dppi_init()
```

Comment:
The function signature should be explicit to indicate that the function takes no parameters
```suggestion
void nrf_dppi_init(void)
```